src/main.rs

```
@@ -221,27 +519,105 @@ fn main() -> Result<(), Box<dyn Error>> {
         .iter()
         .map(|layer_folder| format!("{}{}", base_path.clone(), layer_folder))
         .collect();
+
     let ordered_traits = compare_and_verify_traits::<String>(traits, traits_by_config)?;
 
-    let layers = get_layers_by_traits(ordered_traits);
+    let all_layers = get_layers_by_traits(ordered_traits);
+    let mut permutations: HashMap<u64, Vec<String>> = HashMap::new();
+    let mut new_all_layers = all_layers.clone();
+    let mut possible_permutations: usize = 0;
+    let mut no_of_layers: usize = 0;
+    let mut rest_of_items_percentage = config.total_supply;
+
+    // Luego, puedes llamar a la función validate_percentage_sum:
+    match validate_percentage_sum(&config.forced_combinations) {
+        Ok(()) => println!("Percentages are OK."),
+        Err(err) => println!("Error: {}", err),
```

Maybe we can throw a custom error and stop the app.